### PR TITLE
Add markdownlint, link, anchor checks

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? (Required)

add the same checks from https://github.com/pingcap/docs

**Repo admin might need to enable the CI check and mark it "required" in the settings.**

to close https://github.com/tidbcloud/dbaas-docs/issues/14, https://github.com/tidbcloud/dbaas-docs/issues/15, https://github.com/tidbcloud/dbaas-docs/issues/16

@iamxy FYI.

### CODE REVIEW

**Comment by yikeke** [comment]
cc @YiniXu9506 , @aylei, @cofyc

File: scripts/generate_pdf.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+--template=templates/template.tex \
+--columns=80 \
+--listings \
+-V title="TiDB Documentation" \
+-V author="PingCAP Inc." \
+-V date="${_version_tag}" \
+-V CJKmainfont="${MAINFONT}" \
+-V fontsize=12pt \
+-V geometry:margin=1in \
+-V include-after="\\input{templates/copyright.tex}" \
```
Comment (YiniXu9506) [inline scripts/generate_pdf.sh:28]: Also need to add templates to generate PDF. Refer to https://github.com/pingcap/docs-dm/tree/master/templates

Comment (yikeke) [reply scripts/generate_pdf.sh:28]: done

Comment (YiniXu9506) [reply scripts/generate_pdf.sh:28]: The PDF generation part LGTM.

**Review by aylei — APPROVED** [review]
lgtm

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+         command: |
+           scripts/verify-links.sh
+      - run:
+         name: "Check link anchors"
+         command: |
+           scripts/verify-link-anchors.sh
+      - run:
+         name: "Check unclosed tags"
+         command:
+          python3 scripts/check-tags.py $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md' ':(exclude).github/*')
```
Comment (yikeke) [inline .circleci/config.yml:39]:
```suggestion
           python3 scripts/check-tags.py $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md' ':(exclude).github/*')
```

**Comment by yikeke** [comment]
@iamxy Could you please help configure circleci for this repo?

**Pull request merged**